web/services/TemplateService.php

```diff
@@ -0,0 +1,232 @@
+<?php
+/*
+ * Copyright (C) 2021 Igalia, S.L. <info@igalia.com>
+ *
+ * This file is part of PhpReport.
+ *
+ * PhpReport is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PhpReport is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PhpReport.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace Phpreport\Web\Services;
+
+if (!defined('PHPREPORT_ROOT')) define('PHPREPORT_ROOT', __DIR__ . '/../../');
+
+include_once(PHPREPORT_ROOT . '/web/services/WebServicesFunctions.php');
+include_once(PHPREPORT_ROOT . '/model/facade/TemplatesFacade.php');
+
+class TemplateService
+{
+    private \TemplateVO $templateVO;
+    private \LoginManager $loginManager;
+
+    public function __construct(
+        \TemplateVO $templateVO,
```

Comment:
This will create an object every time, but now it is created outside the class. The idea is to use dependency injection to make the class more independent and easier to isolate and test. 

But for TemplateVo you are probably right that I don't need to do this since it's very isolated already, I'll change this and pass only the LoginManager.

Reply:
I've been reading about dependency injection and, while I can see the advantage of injecting the LoginManager, I'm not sure about the TemplateVO. I wonder what's the usual procedure: inject all classes, decide case-by-case... The main difference for me is that, while LoginManager provides a service, TemplateVO acts more like a basic data type.

> This will create an object every time, but now it is created outside the class. 

I see, now I realize the template create operation only supports creating one template for every operation. But, I wonder how DI could be achieved if the service supported creating several at the same time, like the task create service does.